plugins/repository-s3/qa/amazon-s3/build.gradle

```diff
@@ -39,16 +39,37 @@ forbiddenApisTest {
 
 boolean useFixture = false
 
-String s3AccessKey = System.getenv("amazon_s3_access_key")
-String s3SecretKey = System.getenv("amazon_s3_secret_key")
-String s3Bucket = System.getenv("amazon_s3_bucket")
-String s3BasePath = System.getenv("amazon_s3_base_path")
-
-if (!s3AccessKey && !s3SecretKey && !s3Bucket && !s3BasePath) {
-    s3AccessKey = 's3_integration_test_access_key'
-    s3SecretKey = 's3_integration_test_secret_key'
-    s3Bucket = 'bucket_test'
-    s3BasePath = 'integration_test'
+// We test against two repositories, one which uses the usual two-part "permanent" credentials and
+// the other which uses three-part "temporary" or "session" credentials.
+
+String s3PermanentAccessKey = System.getenv("amazon_s3_access_key_permanent")
+String s3PermanentSecretKey = System.getenv("amazon_s3_secret_key_permanent")
+String s3PermanentBucket = System.getenv("amazon_s3_bucket_permanent")
+String s3PermanentBasePath = System.getenv("amazon_s3_base_path_permanent")
+
+String s3TemporaryAccessKey = System.getenv("amazon_s3_access_key_temporary")
+String s3TemporarySecretKey = System.getenv("amazon_s3_secret_key_temporary")
+String s3TemporarySessionToken = System.getenv("amazon_s3_session_token_temporary")
+String s3TemporaryBucket = System.getenv("amazon_s3_bucket_temporary")
+String s3TemporaryBasePath = System.getenv("amazon_s3_base_path_temporary")
+
+// If all these variables are missing then we are testing against the internal fixture instead, which has the following
+// credentials hard-coded in.
+
+if (!s3PermanentAccessKey && !s3PermanentSecretKey && !s3PermanentBucket && !s3PermanentBasePath
```

Comment:
Can we set the defaults in separate condition blocks? So that if a set [amazon_s3_access_key/amazon_s3_secret_key/amazon_s3_base_path/amazon_s3_bucket] is set it runs against the real service whatever is set for `amazon_s3_access_key_for_session_tokens`

Reply:
As it currently stands, I think it makes more sense to either test entirely against the fake fixture or entirely against the real service (and therefore not run the fake fixture). If there is a followup that allows these two cases to run separately then hopefully it's not too hard to disentangle this then.